moabb/evaluations/splitters.py

```diff
@@ -267,3 +267,125 @@ def split(self, y, metadata):
+        Pass an int for reproducible output across multiple function calls.
+        For `cv_class` accepting `random_state`, they are provided with a shared rng.
+    cv_kwargs: dict
+        Additional arguments to pass to the inner cross-validation strategy.
+
+    Yields
+    ------
+    train : ndarray
+        The training set indices for that split.
+
+    test : ndarray
+        The testing set indices for that split.
+    """
+
+    def __init__(
+        self,
+        cv_class: type[BaseCrossValidator] = LeaveOneGroupOut,
+        shuffle: bool = False,
+        random_state: int = None,
+        **cv_kwargs,
+    ):
+        self.cv_class = cv_class
+        self.cv_kwargs = cv_kwargs
+        self._cv_kwargs = dict(**cv_kwargs)
+
+        self.shuffle = shuffle
+        self.random_state = random_state
+
+        params = inspect.signature(self.cv_class).parameters
+        if shuffle and ("shuffle" not in params and "random_state" not in params):
+            raise ValueError(
+                f"Shuffling is not supported for {cv_class.__name__}. "
+                "Choose a different `cv_class` or use `shuffle=False`."
+                "Example of `cv_class`: `GroupShuffleSplit`: "
+                "CrossSubjectSplitter(shuffle=True, random_state=42, cv_class=GroupShuffleSplit)"
+            )
+
+        if not shuffle and "shuffle" in params and random_state is not None:
+            raise ValueError(
+                "`random_state` should be None when `shuffle` is False for {cv_class.__name__}"
+            )
+
+        self._need_rng = "random_state" in params and (shuffle or "shuffle" not in params)
+
+        if "shuffle" in params:
+            self._cv_kwargs["shuffle"] = shuffle
+
+    def get_n_splits(self, metadata):
+        """
+        Return the number of splits for the cross-validation.
+
+        The number of splits is the number of subjects times the number of splits
+        of the inner cross-validation strategy.
+
+        We try to keep the same behaviour as the sklearn cross-validation classes.
+
+        Parameters
+        ----------
+        metadata: pd.DataFrame
+            The metadata containing the subject and session information.
+
+        Returns
+        -------
+        n_splits: int
+            The number of splits for the cross-validation
+        """
+
+        splitter = self.cv_class(**self._cv_kwargs)
+        n_splits = splitter.get_n_splits(metadata.index, groups=metadata["subject"])
+        return n_splits
+
+    def split(self, y, metadata):
+        # here, I am getting the index across all the subject
+        all_index = metadata.index.values
+
+        # To make sure that when I shuffle the subject, I shuffle the same way
+        # the session when the object is created
+        cv_kwargs = {**self._cv_kwargs}  # Copy the original kwargs
+        if self._need_rng:
+            cv_kwargs["random_state"] = check_random_state(self.random_state)
```

Comment:
```suggestion
        cv_kwargs = {**self._cv_kwargs}  # Copy the original kwargs
        if self._need_rng:
            cv_kwargs["random_state"] = self.random_state
```
Again a comment on the random state ><

With this cross-subject splitter, we only initialize one cv class. So we don't need to manually initialize the rng. We can leave that to the cv class. 
Also, the `random_state` arg could be added to the `cv_kwargs` dict in the `__init__` as we can simply pass it as-is